### ISSUE TITLE
Add support for secrets that are arrays

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#17213]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Users want to use the secrets loading functionality for Vector configuration fields that expect an array; for example: the `splunk_hec` source `valid_tokens` field. Currently only string secrets are supported.

### Attempted Solutions

An external process like `confd` could be used to manage these sorts of secrets.

### Proposal

Allow for secrets to return arrays. These should be encoded as TOML/JSON/YAML-compatible values (depending on the config file language) when interpolating into the config.

### References

- https://github.com/vectordotdev/vector/issues/17172

### Version

vector 0.29.1

**Comment by MadsRC** [comment]
It would also be nice if it was possible to specify dynamic, multiple usernames and password for Basic Auth with the HTTP Server source

**Comment by jszwedko** [comment]
> It would also be nice if it was possible to specify dynamic, multiple usernames and password for Basic Auth with the HTTP Server source

Agreed. Do you mind opening a separate issue for that though? I think we'd need to modify that configuration to support multiple usernames and passwords first and _then_, combined with this issue, it'd be possible to use secrets to drive that.
